### PR TITLE
blog: security patches for Sep 24

### PR DESCRIPTION
**Pull request by UlisesGascon**
This blog entry should go live on Sept 30 (Monday).

cc: @expressjs/security-wg @expressjs/security-triage

### CODE REVIEW

**Comment by UlisesGascon** [comment]
@chrisdel101 I was not able to find this entry in the preview. Maybe is because has a future date? :thinking:

**Comment by UlisesGascon** [comment]
Hey @lirantal! I will love to get your feedback on the format if possible :)

**Comment by chrisdel101** [comment]
Your right! I guess we didn't test this enough. I'm not sure why immediately but I'm looking into this!

**Comment by chrisdel101** [comment]
So yes, it is because it's future dated. If you want to change the file name date to today to preview it, and then once it's done change it back? Just for now?
We'll need to think about how we want to handle this workflow.

**Review by rxmarbles — COMMENTED** [review]
LGTM

File: _posts/2024-09-30-security-releases.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+- `4.20.0`
+- `5.0.0`
+
+
+This vulnerability was discovered during the [OSTIF audit to Express](https://github.com/expressjs/security-wg/issues/6) and was mitigated by [the Express Securty triage team](https://github.com/expressjs/security-wg?tab=readme-ov-file#security-triage-team).
+
+More details area available in [GHSA-qw6h-vgh9-j6wx](https://github.com/expressjs/express/security/advisories/GHSA-qw6h-vgh9-j6wx)
+
+
+## CVE-2024-43799 (moderate) in Send.
```
Comment (wesleytodd) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
## CVE-2024-43799 (moderate) in `send`.
```

File: _posts/2024-09-30-security-releases.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+**Patched versions**
+- `1.20.3`
+
+This vulnerability was discovered during the [OSTIF audit to Express](https://github.com/expressjs/security-wg/issues/6) and was mitigated by [the Express Securty triage team](https://github.com/expressjs/security-wg?tab=readme-ov-file#security-triage-team).
+
+More details area available in [GHSA-qwcr-r2fm-qrc7](https://github.com/expressjs/body-parser/security/advisories/GHSA-qwcr-r2fm-qrc7)
+
+
+
+## CVE-2024-43796 (moderate) in Express
```
Comment (wesleytodd) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
## CVE-2024-43796 (moderate) in `express`
```

File: _posts/2024-09-30-security-releases.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+**Patched versions**
+- `0.19.0`
+
+This vulnerability was discovered during the [OSTIF audit to Express](https://github.com/expressjs/security-wg/issues/6) and was mitigated by [the Express Securty triage team](https://github.com/expressjs/security-wg?tab=readme-ov-file#security-triage-team).
+
+More details area available in [GHSA-m6fv-jmcg-4jfg](https://github.com/pillarjs/send/security/advisories/GHSA-m6fv-jmcg-4jfg)
+
+
+## CVE-2024-43800 (moderate) in Serve-static
```
Comment (wesleytodd) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
## CVE-2024-43800 (moderate) in `serve-static`
```

File: _posts/2024-09-30-security-releases.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+**Patched versions**
+- `>=1.16.0`
+- `>=2.1.0`
+
+This vulnerability was discovered during the [OSTIF audit to Express](https://github.com/expressjs/security-wg/issues/6) and was mitigated by [the Express Securty triage team](https://github.com/expressjs/security-wg?tab=readme-ov-file#security-triage-team).
+
+More details area available in [GHSA-cm22-4g7w-348p](https://github.com/expressjs/serve-static/security/advisories/GHSA-cm22-4g7w-348p)
+
+
+## CVE-2024-45296 (moderate) in Path-to-regexp
```
Comment (wesleytodd) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
## CVE-2024-45296 (moderate) in `path-to-regexp`
```

File: _posts/2024-09-30-security-releases.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
+---
+title: September 2024 Security Releases! 
+tags: security, vulnerabilities
+author: Ulises gascón
+---
+
+Recently, the Express team has been made aware of a number of security vulnerabilities in the Express project.  We have released a number of patches to address these vulnerabilities.  We strongly recommend that all users of Express upgrade to the latest version as soon as possible.  The following vulnerabilities have been addressed:
+
+
+## CVE-2024-45590 (High) in body-parser
```
Comment (wesleytodd) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
## CVE-2024-45590 (High) in `body-parser`
```

**Review by wesleytodd — APPROVED** [review]

File: _posts/2024-09-30-security-releases.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
+---
+title: September 2024 Security Releases!
```
Comment (bjohansebas) [inline _posts/2024-09-30-security-releases.md]: Can you add the 'description' metadata? The page now supports that function.
```suggestion
title: September 2024 Security Releases! 
description: <description>
```

File: _posts/2024-09-30-security-releases.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+**body-parser vulnerable to denial of service when url encoding is enabled**
+
+body-parser `<1.20.3` is vulnerable to denial of service when url encoding is enabled. a malicious actor using a specially crafted payload could flood the server with a large number of requests, resulting in denial of service.
+
+
+**Affected versions**
+- `<1.20.3`
+
+**Patched versions**
+- `1.20.3`
```
Comment (lirantal) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
- `>=1.20.3`
```

File: _posts/2024-09-30-security-releases.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+tags: security, vulnerabilities
+author: Ulises gascón
+---
+
+Recently, the Express team has been made aware of a number of security vulnerabilities in the Express project.  We have released a number of patches to address these vulnerabilities.  We strongly recommend that all users of Express upgrade to the latest version as soon as possible.  The following vulnerabilities have been addressed:
+
+
+## CVE-2024-45590 (High) in `body-parser`
+
+**body-parser vulnerable to denial of service when url encoding is enabled**
```
Comment (lirantal) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
**[body-parser](https://www.npmjs.com/package/body-parser) vulnerable to denial of service when url encoding is enabled**
```

File: _posts/2024-09-30-security-releases.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+This vulnerability was discovered during the [OSTIF audit to Express](https://github.com/expressjs/security-wg/issues/6) and was mitigated by [the Express Securty triage team](https://github.com/expressjs/security-wg?tab=readme-ov-file#security-triage-team).
+
+More details area available in [GHSA-qwcr-r2fm-qrc7](https://github.com/expressjs/body-parser/security/advisories/GHSA-qwcr-r2fm-qrc7)
+
+
+
+## CVE-2024-43796 (moderate) in `express`
+
+**express vulnerable to XSS via `response.redirect()`**
```
Comment (lirantal) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
**[express](https://www.npmjs.com/package/express) vulnerable to XSS via `response.redirect()`**
```

File: _posts/2024-09-30-security-releases.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+In express <4.20.0, passing untrusted user input - even after sanitizing it - to `response.redirect()` may execute untrusted code
```
Comment (lirantal) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
In express <4.20.0, passing untrusted user input - even after sanitizing it - to `response.redirect()` may execute untrusted code.
```

File: _posts/2024-09-30-security-releases.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+**Affected versions**
+- `<4.20.0`
+- `>=5.0.0-alpha.1`, `<5.0.0`
+
+**Patched versions**
+- `4.20.0`
+- `5.0.0`
```
Comment (lirantal) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
- `>=4.20.0`
- `>=5.0.0`
```

File: _posts/2024-09-30-security-releases.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+
+This vulnerability was discovered during the [OSTIF audit to Express](https://github.com/expressjs/security-wg/issues/6) and was mitigated by [the Express Securty triage team](https://github.com/expressjs/security-wg?tab=readme-ov-file#security-triage-team).
+
+More details area available in [GHSA-qw6h-vgh9-j6wx](https://github.com/expressjs/express/security/advisories/GHSA-qw6h-vgh9-j6wx)
+
+
+## CVE-2024-43799 (moderate) in `send`.
+
+**Template injection that can lead to XSS**
```
Comment (lirantal) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
**[send](https://www.npmjs.com/package/send) is vulnerable to template injection that can lead to XSS**
```

File: _posts/2024-09-30-security-releases.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+Passing untrusted user input - even after sanitizing it - to `SendStream.redirect()` may execute untrusted code
+
+
+**Affected versions**
+- `< 0.19.0`
+
+**Patched versions**
+- `0.19.0`
```
Comment (lirantal) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
- `>=0.19.0`
```

File: _posts/2024-09-30-security-releases.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+This vulnerability was discovered during the [OSTIF audit to Express](https://github.com/expressjs/security-wg/issues/6) and was mitigated by [the Express Securty triage team](https://github.com/expressjs/security-wg?tab=readme-ov-file#security-triage-team).
+
+More details area available in [GHSA-m6fv-jmcg-4jfg](https://github.com/pillarjs/send/security/advisories/GHSA-m6fv-jmcg-4jfg)
+
+
+## CVE-2024-43800 (moderate) in `serve-static`
+
+**Template injection that can lead to XSS**
```
Comment (lirantal) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
**[serve-static](https://www.npmjs.com/package/serve-static) vulnerable to template injection that can lead to XSS**
```

File: _posts/2024-09-30-security-releases.md [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+
+Passing untrusted user input - even after sanitizing it - to `redirect()` may execute untrusted code
```
Comment (lirantal) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
Passing untrusted user input - even after sanitizing it - to `redirect()` may execute untrusted code.
```

File: _posts/2024-09-30-security-releases.md [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+- `>=2.1.0`
+
+This vulnerability was discovered during the [OSTIF audit to Express](https://github.com/expressjs/security-wg/issues/6) and was mitigated by [the Express Securty triage team](https://github.com/expressjs/security-wg?tab=readme-ov-file#security-triage-team).
+
+More details area available in [GHSA-cm22-4g7w-348p](https://github.com/expressjs/serve-static/security/advisories/GHSA-cm22-4g7w-348p)
+
+
+## CVE-2024-45296 (moderate) in `path-to-regexp`
+
+**Backtracking regular expressions cause ReDoS**
```
Comment (lirantal) [inline _posts/2024-09-30-security-releases.md]:
```suggestion
**[path-to-regexp](https://www.npmjs.com/package/path-to-regexp) vulnerable to regular expression denial of service (ReDoS)**
```

File: _posts/2024-09-30-security-releases.md [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+- `>=0.2.0`, `<1.9.0`
+- `<0.1.10`
+- `>=2.0.0`, `<3.3.0`
+
+
+**Patched versions**
+- `8.0.0`
+- `1.9.0`
+- `0.1.10`
+- `3.3.0`
```
Comment (lirantal) [inline _posts/2024-09-30-security-releases.md]: these versions should be fixed to provide appropriate version ranges between the different majors.

**Review by lirantal — COMMENTED** [review]
Ran a first pass on the doc.

File: _posts/2024-09-30-security-releases.md (hunk 17 above)
Comment (UlisesGascon) [reply _posts/2024-09-30-security-releases.md]: Added in 33bb348

File: _posts/2024-09-30-security-releases.md (hunk 6 above)
Comment (UlisesGascon) [reply _posts/2024-09-30-security-releases.md]: Added in 33bb348

**Comment by crandmck** [comment]
@UlisesGascon I edited the post for grammar, spelling, and formatting.  I changed the file name to 9/29 so we can see it in the deploy preview.  In your morning it will be 9/30 anyway, but I wanted to be sure the preview looked OK, since I won't be able to work on it on Monday.

This should be good to go.

**Comment by UlisesGascon** [comment]
The last security release expected was added, so this PR is ready for final review. Publication date is expected today.

**Review by carpasse — APPROVED** [review]

File: _posts/2024-09-29-security-releases.md [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+- `>=0.2.0`, `<1.9.0`
+- `<0.1.10`
+- `>=2.0.0`, `<3.3.0`
+
+
+**Patched versions**:
+- `>8.0.0`
+- `>1.9.0`
+- `>=0.1.10`
+- `>3.3.0`
```
Comment (carpasse) [inline _posts/2024-09-29-security-releases.md]: Shouldn't it be:
```
- `>=8.0.0`
- `>=1.9.0`
- `>=0.1.10`
- `>=3.3.0`
```

**Review by carpasse — CHANGES_REQUESTED** [review]

File: _posts/2024-09-29-security-releases.md (hunk 18 above)
Comment (UlisesGascon) [reply _posts/2024-09-29-security-releases.md]: Updated in [bcd9dd3](https://github.com/expressjs/expressjs.com/pull/1607/commits/bcd9dd31f58f927365fa053a5050f6fcf1bd5a68), based on https://github.com/pillarjs/path-to-regexp/security/advisories/GHSA-9wv6-86v2-598j

**Review by carpasse — APPROVED** [review]

**Pull request merged**